lib/migration.c

```diff
@@ -120,7 +140,7 @@ init_migration(const vfu_migration_t * const vfu_migr, int *err)
 static bool
 _migr_state_transition_is_valid(__u32 from, __u32 to)
 {
-    return migr_states[from] & (1 << to);
+    return migr_states[from].state & (1 << to);
```

Comment:
I just noticed this now: don't we want an equality comparison here?
It's not valid to migrate from VFIO_DEVICE_STATE_STOP to
VFIO_DEVICE_STATE_STOP | VFIO_DEVICE_STATE_RUNNING
but a simple bitwise and allows it.

Reply:
No, it's correct the way it is. The destination state sets only one bit in the `state` bit array. I've implemented a unit test, will push tomorrow.

Reply:
Oh, I mis-read the existing code, I missed that you were shifting 1 by an OR'd value